--- a/src/plugins/platforms/windows/qwindowsscreen.cpp	2023-08-24 10:26:41
+++ b/src/plugins/platforms/windows/qwindowsscreen.cpp	2023-08-24 10:30:17
@@ -244,7 +244,7 @@
         names << QString::fromWCharArray(deviceName.monitorFriendlyDeviceName);
         manufacturers << edid.manufacturer;
         models << edid.model;
-        serialNumbers << edid.serialNumber;
+        serialNumbers << edid.serialNumber + data.deviceName;
     }
 
     data.name = names.join(u"|"_s);
