--- configure~	2018-12-01 09:36:00.000000000 -0500
+++ configure	2019-03-29 09:19:06.000000000 -0400
@@ -13136,9 +13136,9 @@
 $as_echo "$as_me: WARNING: not building timepng" >&2;}
 fi
 
- if test "$ac_cv_func_clock_gettime" = "yes"; then
-  HAVE_CLOCK_GETTIME_TRUE=
-  HAVE_CLOCK_GETTIME_FALSE='#'
+if test "$ac_cv_func_clock_gettime" = "yes"; then
+  HAVE_CLOCK_GETTIME_TRUE='#'
+  HAVE_CLOCK_GETTIME_FALSE=
 else
   HAVE_CLOCK_GETTIME_TRUE='#'
   HAVE_CLOCK_GETTIME_FALSE=
--- pngrutil.c~	2018-12-01 09:36:00.000000000 -0500
+++ pngrutil.c	2019-03-29 09:11:26.000000000 -0400
@@ -422,12 +422,14 @@
             png_ptr->flags |= PNG_FLAG_ZSTREAM_INITIALIZED;
       }
 
+#if 0
 #if ZLIB_VERNUM >= 0x1290 && \
    defined(PNG_SET_OPTION_SUPPORTED) && defined(PNG_IGNORE_ADLER32)
       if (((png_ptr->options >> PNG_IGNORE_ADLER32) & 3) == PNG_OPTION_ON)
          /* Turn off validation of the ADLER32 checksum in IDAT chunks */
          ret = inflateValidate(&png_ptr->zstream, 0);
 #endif
+#endif
 
       if (ret == Z_OK)
          png_ptr->zowner = owner;
