--- .//qtbase/src/gui/kernel/qwindow.cpp.orig	2019-12-12 09:15:59.000000000 -0500
+++ .//qtbase/src/gui/kernel/qwindow.cpp	2019-12-14 13:38:35.000000000 -0500
@@ -219,11 +219,22 @@
     if (!QGuiApplicationPrivate::is_app_closing)
         QGuiApplicationPrivate::instance()->modalWindowList.removeOne(this);
 
+#if 0  // Disable this workaround to allow synchronize activate path to work
+
     // focus_window is normally cleared in destroy(), but the window may in
     // some cases end up becoming the focus window again. Clear it again
     // here as a workaround. See QTBUG-75326.
+
     if (QGuiApplicationPrivate::focus_window == this)
         QGuiApplicationPrivate::focus_window = 0;
+
+    // added by me to prevent crashes in showModalWindow on macOS
+    if (QGuiApplicationPrivate::currentMouseWindow == this)
+        QGuiApplicationPrivate::currentMouseWindow = 0;
+
+    if (QGuiApplicationPrivate::currentMousePressWindow == this)
+        QGuiApplicationPrivate::currentMousePressWindow = 0;
+#endif
 }
 
 void QWindowPrivate::init(QScreen *targetScreen)
