--- qtbase/src/gui/kernel/qwindow.cpp.orig	2019-12-12 09:15:59.000000000 -0500
+++ qtbase/src/gui/kernel/qwindow.cpp	2020-01-10 10:36:53.000000000 -0500
@@ -218,12 +218,6 @@
     QGuiApplicationPrivate::window_list.removeAll(this);
     if (!QGuiApplicationPrivate::is_app_closing)
         QGuiApplicationPrivate::instance()->modalWindowList.removeOne(this);
-
-    // focus_window is normally cleared in destroy(), but the window may in
-    // some cases end up becoming the focus window again. Clear it again
-    // here as a workaround. See QTBUG-75326.
-    if (QGuiApplicationPrivate::focus_window == this)
-        QGuiApplicationPrivate::focus_window = 0;
 }
 
 void QWindowPrivate::init(QScreen *targetScreen)
