--- qtbase/src/platformsupport/fontdatabases/mac/qcoretextfontdatabase.mm.orig	2018-10-29 10:20:59.000000000 -0400
+++ qtbase/src/platformsupport/fontdatabases/mac/qcoretextfontdatabase.mm	2018-10-29 10:23:28.000000000 -0400
@@ -331,9 +331,9 @@
 
     if (styles) {
         if (CFNumberRef weightValue = (CFNumberRef) CFDictionaryGetValue(styles, kCTFontWeightTrait)) {
-            float normalizedWeight;
-            if (CFNumberGetValue(weightValue, kCFNumberFloatType, &normalizedWeight))
-                fd->weight = QCoreTextFontEngine::qtWeightFromCFWeight(normalizedWeight);
+            double normalizedWeight;
+            if (CFNumberGetValue(weightValue, kCFNumberFloat64Type, &normalizedWeight))
+                fd->weight = QCoreTextFontEngine::qtWeightFromCFWeight(float(normalizedWeight));
         }
         if (CFNumberRef italic = (CFNumberRef) CFDictionaryGetValue(styles, kCTFontSlantTrait)) {
             double d;
