--- .//Source/cmake/OptionsQt.cmake~	2018-08-03 12:49:38.000000000 -0400
+++ .//Source/cmake/OptionsQt.cmake	2018-08-04 19:47:35.000000000 -0400
@@ -252,7 +252,7 @@
 WEBKIT_OPTION_DEFINE(ENABLE_X11_TARGET "Whether to enable support for the X11 windowing target." PUBLIC ${ENABLE_X11_TARGET_DEFAULT})
 
 option(GENERATE_DOCUMENTATION "Generate HTML and QCH documentation" OFF)
-cmake_dependent_option(ENABLE_TEST_SUPPORT "Build tools for running layout tests and related library code" ON
+cmake_dependent_option(ENABLE_TEST_SUPPORT "Build tools for running layout tests and related library code" OFF
                                            "DEVELOPER_MODE" OFF)
 option(USE_STATIC_RUNTIME "Use static runtime (MSVC only)" OFF)
 
